src/plugins/dashboard/public/dashboard_container/embeddable/dashboard_container.tsx

```diff
@@ -957,7 +958,16 @@ export class DashboardContainer
     for (const panelId of Object.keys(currentChildren)) {
       if (this.getInput().panels[panelId]) {
         const child = currentChildren[panelId];
-        if (apiPublishesUnsavedChanges(child)) child.resetUnsavedChanges();
+        if (apiPublishesUnsavedChanges(child)) {
+          const success = child.resetUnsavedChanges();
+          if (!success) {
+            coreServices.notifications.toasts.addWarning(
```

Comment:
Great to have a warning in this case. I wonder if we should clarify that the user needs to start a new session (open a new tab etc) to get their last saved state back?

Reply:
I did not want to complicate the error message. Its complex because if its a by-ref panel then a new session will not revert the changes because the saved object has already been saved

Reply:
> because the by-ref saved object has already been saved

Yet another example of why we need to fix this architecturally, as `save and return` on a by reference panel shouldn't even show `unsaved changes` on the Dashboard.

++ to not complicating the error message. Ideally this isn't seen by users at all.